server/src/test/java/org/opensearch/cluster/routing/allocation/AwarenessAllocationTests.java

```diff
@@ -1063,4 +1063,39 @@ public void testAllocationExplainForUnassignedShardsWithUnbalancedZones() {
 
         }
     }
+
+    public void testIgnoredByAutoExpandReplicasToAll() {
+        final Settings settings = Settings.builder()
+            .put(AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING.getKey(), "zone")
+            .build();
+
+        final AllocationService strategy = createAllocationService(settings);
+
+        final IndexMetadata.Builder metadataBuilder = IndexMetadata.builder("test")
+            .settings(
+                settings(Version.CURRENT).put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 100)
```

Comment:
If there are 100 replicas and 4 nodes i.e. 101 shards of the index and if no same shard can be assigned on the same node then how come your test passes with no unassign shard because you are only ignoring zone awareness which means you can have at max 4 shards (1 primary and 3 replica assigned of the test index) and other shards should still stay unassigned right?

Reply:
it's Allocation Awareness Test, i assume it would not cover Same Host allocation scenario. as it ignores the zone awareness, so all shard would be assigned.

Reply:
oh, Can you please add an IT as well for your use case?